song/manifest.go

```diff
@@ -0,0 +1,55 @@
+/*
+ *     Copyright (C) 2018  Ontario Institute for Cancer Research
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Affero General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Affero General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Affero General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package song
+
+import (
+	"encoding/json"
+)
+
+type manifestFile struct {
+	Info       map[string]string
+	ObjectId   string
+	AnalysisId string
+	StudyId    string
+	FileName   string
+	FileSize   int64
+	FileType   string
+	FileAccess string
+	FileMd5sum string
+}
+
+func (f *manifestFile) String() string {
+	return f.ObjectId + "\t" + f.FileName + "\t" + f.FileMd5sum
+	return ""
+}
+
+func createManifest(analysisID string, data string) string {
+	var files []manifestFile
+
+	err := json.Unmarshal([]byte(data), &files)
+	if err != nil {
+		panic("Can't unmarshal '" + data + "'")
```

Comment:
I know GO calls it "unmarshal" for parsing JSON but an end-user might not understand that error message unless they have Go experience, might be worth making this error message a little more generic?

Reply:
Hmm... if they ask for a manifest for an analysis that doesn't exist, then they **could** get JSON that isn't valid for a manifest... we should handle that case, and report the message that the SONG server returns.

Otherwise, about all we could say would be: "Our SONG server returned invalid JSON for your manifest file; this isn't your fault, and you can't fix it, so please email us to let us know something is wrong?" :-)